sphinx/roles.py

```diff
@@ -241,6 +241,17 @@ def build_uri(self) -> str:
             return base_url + self.inliner.rfc_url % int(ret[0])
 
 
+def _format_rfc_target(target: str) -> str:
+    """
+    Takes an RFC number with an optional anchor (like ``123#section-2.5.3``) and returns
+    a nicely formatted title for it.
+    """
+    parts = target.replace('#', ' ').replace('-', ' ').split()
+    if len(parts) >= 2:
+        parts[1] = parts[1].title()
```

Comment:
RFC links can have section names, for example

https://datatracker.ietf.org/doc/html/rfc9076.html#name-risks-in-the-dns-data

So you should only do this formatting for exact matches for a few specific cases, I think the ones you put in the test "Section" "Page" "Appendix" are good.

Reply:
Right, I didn't know that was possible – thank you.

What do you think we should do in that case? Leaving it be and rendering as `RFC 9076#name-risks-in-the-dns-data` is easy but it doesn't look great.

On the other hand I can't think of a good capitalization logic that would convert things like "risks-in-the-dns-data" to "Risks in the DNS data" reliably.

Something like this would be easy (when a `name-` prefix is detected in the anchor): `RFC 9076 Section risks-in-the-dns-data`.

Reply:
PS. 

> the above would be formatted as "Name Risks in the Dns Data"

Note that I'm currenly applying `title()` to the first word in that sequence, not the whole thing.

Reply:
> good capitalization logic

If we're only limiting ourselves to links that appear on datatracker.ietf.org then as I said, we have a specific number of text documents (RFCs) with a specific number of human readable section names that are being lowercased and `.replace(' ', '-')`'d and we want to recover those specific set of original strings. The way we can do it is to extract every single possible link on the website and the actual original human readable text, which should be doable locally (i.e. without scraping) (I did something similar once [here](https://github.com/m4tx/py3redirect/pull/7) and [here](https://github.com/python/cpython/pull/24195)) since it's an open source website. There's like 9000 RFCs, so that's maybe a list of 900,000 2 item test cases that your function needs to convert correctly. Title casing (without "and" "as" "but" "for" "if" "nor" "or" "so" "yet" "a" "an" "the" "as" "at" "by" "for" "in" "of" "off" "on" "per" "to" "up" "via") + a list of acronyms + 100 hard coded special cases would probably do it. Then you make the extraction easily repeatable so that people can periodically re-run the extraction code as the RFCs are edited. Also, I think for section names it might make sense to also add a colon like "RFC 9076: Name Risks in the DNS Data".

Or you can just do it for sections, to me it's better in URL syntax because it's a link.

> Note that I'm currenly applying title() to the first word in that sequence, not the whole thing.

Right, I forgot what `.title()` does.

Reply:
I decided to leave anchors with names (and anything else not already determined to be safe) alone for now, probably not worth blocking this PR because of that.

Reply:
Ideally, 

- RFC 1234, Section 2.
- RFC 1234, Appendix 2, page 23.
- RFC 1234: The Section Title (capitalize words except common conjunctions)

But I'm not sure whether it's worth the case-handling.